src/reports/library/reports.cpp

```diff
@@ -0,0 +1,196 @@
+#include "../utils/logger.h"
+#include "sonatareport.h"
+#include <bbp/sonata/reports.h>
+#include <iostream>
+
+using namespace bbp::sonata;
+
+SonataReport sonata_report;
```

Comment:
having static objects scares me, is this required?

Reply:
Is the best way I came up to call the C++ API from the C wrapper.
Another option would be to create structs to imitate the classes (something like https://stackoverflow.com/questions/2045774/developing-c-wrapper-api-for-object-oriented-c-code)
But I thought it was really convoluted solution

Reply:
AFAIK, that's the best practice for wrapping `C++`; having a static means you can run into initialization ordering issues, multithreaded isses, and when you write tests, you have to reset everything each time.

Reply:
Looking at the failing build, it seems that clang is unhappy w/o the dtor:
https://travis-ci.com/BlueBrain/libsonata/jobs/288530337#L454
```
In file included from /Users/travis/build/BlueBrain/libsonata/src/reports/library/report.cpp:2:
In file included from /Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/iostream:38:
In file included from /Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/ios:216:
In file included from /Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__locale:15:
In file included from /Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:500:
In file included from /Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:176:
In file included from /Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:644:
/Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:3706:5: error: destructor called on non-final 'bbp::sonata::Node' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-virtual-dtor]
    __data_.second().~_Tp();
    ^
/Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:3667:9: note: in instantiation of member function 'std::__1::__shared_ptr_emplace<bbp::sonata::Node, std::__1::allocator<bbp::sonata::Node> >::__on_zero_shared' requested here
        __shared_ptr_emplace(_Alloc __a, _Args&& ...__args)
        ^
/Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:4327:26: note: in instantiation of function template specialization 'std::__1::__shared_ptr_emplace<bbp::sonata::Node, std::__1::allocator<bbp::sonata::Node> >::__shared_ptr_emplace<unsigned long long &>' requested here
    ::new(__hold2.get()) _CntrlBlk(__a2, _VSTD::forward<_Args>(__args)...);
                         ^
/Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:4706:29: note: in instantiation of function template specialization 'std::__1::shared_ptr<bbp::sonata::Node>::make_shared<unsigned long long &>' requested here
    return shared_ptr<_Tp>::make_shared(_VSTD::forward<_Args>(__args)...);
                            ^
/Users/travis/build/BlueBrain/libsonata/src/reports/library/report.cpp:35:38: note: in instantiation of function template specialization 'std::__1::make_shared<bbp::sonata::Node, unsigned long long &>' requested here
        nodes->emplace(node_id, std::make_shared<Node>(node_id));
                                     ^
/Applications/Xcode-10.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:3706:23: note: qualify call to silence this warning
    __data_.second().~_Tp();
```

Reply:
Yep, I saw. Putting back virtual ~Node() = default; and virtual ~Report() = default fixes the complain